### ISSUE TITLE
[GPU Texture Compression] Re: Compression failure due to spaces in folder name [Mac/Linux]

### ISSUE THREAD

**Issue by GameDevTosh** [YoYoGames/GameMaker-Bugs#6180]
### Description

Previously this bug was reported here: https://github.com/YoYoGames/GameMaker-Bugs/issues/5720

It was fixed and closed about a month ago, and I/we confirmed ourselves it worked for Windows. Today, Juju found the bug when building on Mac/Linux. It seems it was only fixed for Windows. Since the old report is listed as closed, I made a new report to get more eyes on it, as I'm unsure if closed reports are noticed.

### Expected Change

Allowed spaces in folder paths, for all platforms.

### Steps To Reproduce

Please see the previous bug report for steps and example project. The only difference would be to test on Linux and Mac.

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.4.1 (Monthly)

### Which operating system(s) are you seeing the problem on?

Windows 10

### Which platform(s) are you having trouble building this project for?

macOS, Ubuntu

### Are you running GameMaker from inside your Steam library?

None

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [ ] I have included a small sample project

**Comment by JujuAdams** [comment]
Yep, issue not fixed for Mac/Linux IDE. Exact same problem, exact same solution - put the input and output path strings inside quotes.

**Comment by JonnoRobson** [comment]
This issue slipped through due to differences in how zsh terminal handles spaces in path names versus bash terminal (previous version of the convTextures script ran in zsh rather than bash but this was changed for other compatibility reasons). I've added quotation marks around all of the path names in the script to fix the issue. There was also a similar issue in the asset compiler due missing quotation marks when trying to use chmod to set the script as executable which I have also fixed - this fix will be available in the next BETA build. Combined these fixes appear to resolve all issues with running the texture encoding extension on Mac when there are spaces in the project path.

**Comment by Emc1923** [comment]
Verified as of IDE v2024.6.0.156 Runtime v2024.6.0.205/Beta IDE v2024.600.0.585 Beta Runtime v2024.600.0.609
